src/librustc_mir/interpret/cast.rs

```diff
@@ -67,7 +67,7 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 // The src operand does not matter, just its type
                 match src.layout.ty.sty {
                     ty::Closure(def_id, substs) => {
-                        let substs = self.subst_and_normalize_erasing_regions(substs)?;
+                        let substs = self.subst_and_normalize_erasing_regions_in_frame(substs)?;
```

Comment:
This is wrong, you got this from a layout, therefore it should already be "in the right universe".